### ISSUE TITLE
Why BatchNorm2D has inconsistent gradients and sizes for running_mean and running_variance

### ISSUE THREAD

**Issue by aSingh071954** [pytorch/pytorch#5742]
Hi, 

I have a question regarding BatchNorm2d, below is the output when i call optimizer.step() to update parameters. I can see that the conv1.weight tensor size is 64 but it should have been 64x64x3x3 (Number of Input Channels = 64)?

Similarly, why bn1.running_mean size is [64, 64, 3, 3] ? 
And why module.bn1.running_var size is [64] ?
Why don't we have Gradients computed for running_mean as we have computed for running_var OR why gradients are computed for running_var at all (as it is not learnable parameter) ?

Below are the outputs I printed for the conv1 and batchnorm layer -->

module.conv1.weight: 0.5900921821594238
module.conv1.weight.size(): torch.Size([64])
module.conv1.weight.Gradients: 0.8657947182655334

module.bn1.running_mean: 1.9200005531311035
module.bn1.running_mean.size(): torch.Size([64, 64, 3, 3])

module.bn1.running_var: 0.6700683832168579
module.bn1.running_var.size(): torch.Size([64])
module.bn1.running_var.Gradients: 2.270807981491089

I am using the following resent model from -->
https://github.com/kuangliu/pytorch-cifar/blob/master/models/resnet.py

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions
